### DISCUSSION TITLE
documentation for fulfillRandomness

### DISCUSSION THREAD

**Discussion by AlbertoSinigaglia** [smartcontractkit/full-blockchain-solidity-course-py#27] [Q&A] [answered]
Hi, I was looking nat [chainlink documentation](https://docs.chain.link/docs/chainlink-vrf-api-reference/#fulfillrandomness) of `fulfillRandomness` that says:
 > randomness: Random number generated by Chainlink VRF.

However in the code shown in the course, we use `randomness == 0` to check if an error is occurred... at the same time, I don't see anywhere something that says that 0 is considered as an "error" for the random generated number... what am I missing? Are there any documentation that warns about this?

**Comment by PatrickAlphaC** [comment, marked as answer]
This is an AMAZING question.

You're right, the RNG could 100% return a random number of 0, however... the chance is very very very very low. We are initializing our random number to 0, so as a quick check to make sure we got a response, we check to see that the number we got isn't 0.

The is a 1 in 2**256 chance that we get 0 back, so it's highly unlikely to happen.

**Reply by AlbertoSinigaglia** [reply]
ok makes sense, 2**256 it's pretty rare ahah but why are we even checking? this is an internal function, so the only way to call it it's from inside, and this function is executed only if we have the cryptographic proof that the request comes from the VRF no? so this function is called only inside, and the only way that out contract is calling it is if it receives a request from a verified VRF... so there is no way that the generation fail no? I mean, if the it fails, then the VRF wouldn't call back our contract

**Reply by PatrickAlphaC** [reply]
You're right again! 

It was more as an educational demonstration... but it looks like it was more confusing than helpful. Thanks for calling it out!
